kernel/aster-nix/src/process/process_vm/heap.rs

```diff
@@ -72,18 +63,23 @@ impl Heap {
                     // FIXME: should we allow shrink current user heap?
                     return Ok(current_heap_end);
                 }
+                let old_size = (current_heap_end - self.base).align_up(PAGE_SIZE);
                 let new_size = (new_heap_end - self.base).align_up(PAGE_SIZE);
-                let heap_mapping = root_vmar.get_vm_mapping(USER_HEAP_BASE)?;
-                let heap_vmo = heap_mapping.vmo();
-                heap_vmo.resize(new_size)?;
+
+                // FIXME: Our current implementation of mapping resize cannot move
```

Comment:
Just a reminder: This problem in this FIXME is severe since our heap is in rather low address. If the user maps some big files, this heap area may be used. So it should be fixed soon.